census/estimator/trainer/model.py

```diff
@@ -278,9 +278,17 @@ def input_fn(filenames,
       A (features, indices) tuple where features is a dictionary of
         Tensors, and indices is a single Tensor of label indices.
   """
-
-  dataset = tf.data.TextLineDataset(filenames).skip(skip_header_lines).map(parse_csv)
-
+  filename_dataset = tf.data.Dataset.from_tensor_slices(filenames)
+  if shuffle:
```

Comment:
@mrry We don't need two if shuffle.

Reply:
According to the doc string (and best practice) you do:

> `shuffle`: bool, whether or not to randomize the order of data. Controls randomization of both file order and line order within files.

The first shuffle shuffles the files, and the second shuffle shuffles the lines within the files.

Reply:
Correct. I overlooked code between two if statements and was wrongly suggesting collapsing ifs. Thanks!